### PR TITLE
task DEV-1

### PR DESCRIPTION
**Pull request by AKhlebko**
Code's ready for review

### CODE REVIEW

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        /// and for for finding the longest same character substring
+        /// Stores processed string in a private field
+        /// </summary>        
+        private string str;
+
+        public int LongestSubStringSameSymbols()
+        {
+            /// <summary>
+            /// Returns length of the longest substring consisting of the same symbols
+            /// </summary>
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: Not right format for methods comments

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            {
+                return 0;
+            }
+            for (int i = 0; i < str.Length - 1; i++)
+            {
+                int tempCounter = 1;
+                int j = i + 1;
+                while (j < str.Length)
+                {
+                    try
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: What try? which statement can throw exception?

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                    }
+                }
+                if (tempCounter > MaxStringLength)
+                {
+                    MaxStringLength = tempCounter;
+                }
+            }
+            return MaxStringLength;
+        }
+
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: Why args? Why massive of strings?

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            if (args.Length == 0)
+            {
+                str = "";
+            }
+            else
+            {
+                GetStringFromArguments(args);
+            }
+        }
+
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: Bad decision, only one argument in console is allowed

File: DEV-1/DEV-1/EntryPoint.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace DevTask1
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/EntryPoint.cs]: If class name is Finder, why value is handler?

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                {
+                    MaxStringLength = tempCounter;
+                }
+            }
+            return MaxStringLength;
+        }
+
+        public SameSymbolSubstringFinder(string[] args)
+        {
+            /// <summary>
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: Before

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                    maxStringLength = tempCounter;
+                }
+            }
+            return maxStringLength;
+        }
+
+        /// <summary>
+        /// StringHandler class's instance initializer
+        /// Gets console args as input arguments
+        /// </summary>
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: According to code style constructors are in top of desription

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    /// <summary>
+    /// This class contains methods for getting string
+    /// and for for finding the longest same character substring
+    /// Stores processed string in a private field
+    /// </summary>
+    class SameSymbolSubstringFinder
+    {
+        private string processedString;
+
+        /// <summary>
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: not right format of comments

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        /// Returns length of the longest substring consisting of the same symbols
+        /// </summary>
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: Better name is GetLongest.... if method return something it has to be clear from the name

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+                    {
+                        j++;
+                        tempCounter++;
+                    }
+                    else
+                    {
+                        i = j - 1;
+                        break;
+                    }
+
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: Empty line

File: DEV-1/DEV-1/SameSymbolSubstringFinder.cs [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+            int maxStringLength = 0;
+            if (processedString.Length == 0)
+            {
+                return 0;
+            }
+            for (int i = 0; i < processedString.Length - 1; i++)
+            {
+                int tempCounter = 1;
+                int j = i + 1;
+                while (j < processedString.Length)
```
Comment (IgorKheidorov) [inline DEV-1/DEV-1/SameSymbolSubstringFinder.cs]: Simplify logics, cycle inside cycle is not a good idea

File: .gitattributes [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+#*.gif   binary
+
+###############################################################################
+# diff behavior for common document formats
+# 
+# Convert binary document formats to text before diffing them. This feature
+# is only available from the command line. Turn it on by uncommenting the 
+# entries below.
+###############################################################################
+#*.doc   diff=astextplain
```
Comment (AlexanderSoroka) [inline .gitattributes]: you don't need that

File: .gitattributes [hunk 13]
```diff
@@ -0,0 +1,63 @@
+###############################################################################
```
Comment (AlexanderSoroka) [inline .gitattributes]: You have just copy-pasted that file, please cleanup and leave only actual parts

File: .gitignore [hunk 14]
```diff
@@ -1,288 +1,5 @@
-## Ignore Visual Studio temporary files, build results, and
```
Comment (AlexanderSoroka) [inline .gitignore]: How did so much removal changes appear in that changeset? Please remove from review everything that doesn't relate to the task

Comment (AKhlebko) [reply .gitignore]: I read your comment and deleted .gitignore and .gitattribute from the branch. I just dunno how to delete the history about them

Comment (AKhlebko) [reply .gitignore]: If you can help me, i'll very appreciate it

**Pull request merged**